index.ts

```diff
@@ -170,15 +191,71 @@ async function main() {
   });
   console.log(safeOpHash);
 
-  const signature = await getNearSignature(safeOpHash);
+  // const signature =
+  //   "0xef028109f94aabaf9cd4fb7f2cce311527567341d3439a40fe15e6e01c45249b790e5e5c1f0803f668219f97a95947a982e4f88d8f9abc675adcdb926ed7504d1c";
+  const signature = await getNearSignature(nearAdapter, safeOpHash);
   console.log(
     await sendUserOperation(
-      { ...unsignedUserOp, signature },
+      asERC4337Payload({ ...unsignedUserOp, signature }),
       await contracts.entryPoint.getAddress(),
     ),
   );
 }
 
+interface UserOpData {
```

Comment:
Nice types! Just going to nit on the names so it's consistent with the ERC: this should be:


```suggestion
interface UserOperation {
```

Reply:
Ya I had a feeling the names would be terrible. You should have seen what I started with "input" and "output"!